### PR TITLE
Add luminance getter to Color

### PR DESCRIPTION
**Pull request by xster**
Will remove from Material's ThemeData after

### CODE REVIEW

**Comment by Hixie** [comment]
tests?

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -159,6 +159,27 @@ class Color {
...
+    return math.pow((component + 0.055) / 1.055, 2.4);
+  }
+
+  /// Returns a brightness value between 0 for darkest and 1 for lightest.
+  ///
+  /// Represents the relative luminance of the color. This value is computationally
+  /// expensive to calculate.
+  ///
+  /// See <https://en.wikipedia.org/wiki/Relative_luminance>.
+  double get luminance {
```
Comment (Hixie) [inline lib/ui/painting.dart]: If it's expensive then we should make this a method like `computeLuminance()`, to make it clearer that this is a non-trivial operation. See our design principles guide for more advice along those lines.

Comment (xster) [reply lib/ui/painting.dart]: Makes sense, done

**Comment by xster** [comment]
Added tests, had trouble finding where the test files were before

File: lib/ui/painting.dart [hunk 2]
```diff
@@ -200,6 +200,27 @@ class Color {
     return new Color.fromARGB(alpha, red, green, b);
   }
 
+    // See <https://www.w3.org/TR/WCAG20/#relativeluminancedef>
```
Comment (Hixie) [inline lib/ui/painting.dart]: nit: your comment is overindented

**Comment by Hixie** [comment]
LGTM

File: lib/ui/painting.dart (hunk 2 above)
Comment (xster) [reply lib/ui/painting.dart]: Done, thanks

**Pull request merged**